slashing-protection/src/main/java/tech/pegasys/web3signer/slashingprotection/SlashingProtectionFactory.java

```diff
@@ -22,16 +23,38 @@
 
 public class SlashingProtectionFactory {
 
+  public static final int EXPECTED_DATABASE_VERSION = 6;
+  private static final String errorMsg =
+      String.format(
+          "Database does not have expected version (%s), please run migrations and try again.",
+          EXPECTED_DATABASE_VERSION);
+
   public static SlashingProtection createSlashingProtection(
       final String slashingProtectionDbUrl,
       final String slashingProtectionDbUser,
       final String slashingProtectionDbPassword) {
     final Jdbi jdbi =
         DbConnection.createConnection(
             slashingProtectionDbUrl, slashingProtectionDbUser, slashingProtectionDbPassword);
+
+    verifyVersion(jdbi);
+
     return createSlashingProtection(jdbi);
   }
 
+  private static void verifyVersion(final Jdbi jdbi) {
+    final DatabaseVersionDao databaseVersionDao = new DatabaseVersionDao();
+
+    try {
+      final Integer version = jdbi.withHandle(databaseVersionDao::findDatabaseVersion);
+      if (version.compareTo(EXPECTED_DATABASE_VERSION) != 0) {
+        throw new IllegalStateException(errorMsg);
+      }
+    } catch (final IllegalStateException e) {
```

Comment:
Is this IllegalStateException is the exception we are throwing on line 51? Looks like it could be.

Reply:
ah - this could be bad, refactored.